libs/clients/inna/src/lib/inna.provider.ts

```diff
@@ -0,0 +1,45 @@
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { Provider } from '@nestjs/common'
+import {
+  ConfigType,
+  IdsClientConfig,
+  LazyDuringDevScope,
+  XRoadConfig,
+} from '@island.is/nest/config'
+import { Configuration, DefaultApi } from '../../gen/fetch'
+import { InnaClientConfig } from './inna.config'
+
+export const InnaClientProvider: Provider<DefaultApi> = {
+  provide: DefaultApi,
+  scope: LazyDuringDevScope,
+  useFactory: (
+    xroadConfig: ConfigType<typeof XRoadConfig>,
+    config: ConfigType<typeof InnaClientConfig>,
+    idsClientConfig: ConfigType<typeof IdsClientConfig>,
+  ) =>
+    new DefaultApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-inna',
+          // organizationSlug: 'menntamalastofnun',
+          logErrorResponseBody: true,
+          /*autoAuth: idsClientConfig.isConfigured
+            ? {
+                mode: 'tokenExchange',
+                issuer: idsClientConfig.issuer,
+                clientId: idsClientConfig.clientId,
+                clientSecret: idsClientConfig.clientSecret,
+                scope: ['@mms.is/inna'],
+              }
+            : undefined,*/
```

Comment:
To be implemented?

Reply:
Yes, the api is being implemented in x-road